Source/items.cpp

```diff
@@ -1740,66 +1740,65 @@ void DrawUniqueInfoWindow(const Surface &out)
 void printItemMiscKBM(const Item &item, const bool isOil, const bool isCastOnTarget)
 {
 	if (item._iMiscId == IMISC_MAPOFDOOM) {
-		AddPanelString(_("Right-click to view"));
+		InfoString = StrCat(item._iName, "\n", _("Right-click to view"));

```

Comment:
What is the purpose of replacing calls to `AddPanelString()` with a multiline `InfoString`? Seems like you're just inviting potential for OOB in `PrintInfo()` by bypassing the validation in `AddPanelString()`. Furthermore, you've changed a few strings that have likely already been translated into many languages so I'd like to understand the justification.

Reply:
The multiline `InfoString` is to allow for multiline strings to be translated in one shot. Right now, for "Right-click to..., then" and "left-click to..." are sent to the translation function separately, which doesn't translate well. I could put the validations into PrintInfo, I could restrict this logic to just times when it's necessary to translate a multiline string, or both.

Reply:
At a bare minimum, we should protect against the OOB. Honestly, though, it wasn't the only red flag here. The fact that you are introducing concatenation with the item name suggests to me that you are overwriting the text that was already assigned to `InfoString` by another part of the code.

I'd rather see a helper that handles multiline strings via `AddPanelString()`. You may also consider breaking this out into another PR in case it generates further discussion and blocks the merge.

Reply:
Actually, I just got a response from AJenbo on another PR so maybe my suggestion to use `AddPanelString()` is not the right approach. See https://github.com/diasurgical/devilutionX/pull/5316#issuecomment-1251655975

Reply:
See [!5267](https://github.com/diasurgical/devilutionX/pull/5267)

Reply:
My last critique in regards to this is that maybe we should stick with `AddPanelString()` rather than replacing the entirety of `InfoString`. This will be especially important for oils, which have text inserted between the item name and the usage instruction.

![image](https://user-images.githubusercontent.com/9203145/194720779-fab0f53a-d33d-4b7a-9b50-989342b27903.png)

Reply:
My current code isn't working, so it needs to change in some way. What if I changed InfoString in PrintItemMiscOil as well, so that everything would still happen in the correct order? What do you think, @AJenbo ?

Reply:
Yeah makes sense now that I can pass a multiline string to AddPanelString.